pyroclast/luigi/train_task.py

```diff
@@ -0,0 +1,46 @@
+import luigi
+import os.path as osp
+import subprocess
+
+
+class TrainTask(luigi.Task):
+    base_path = luigi.Parameter()
+
+    dataset = luigi.Parameter()
+    batch_size = luigi.IntParameter()
+    seed = luigi.IntParameter()
+
+    learning_rate = luigi.FloatParameter()
+    conv_stack_name = luigi.Parameter()
+    lambd = luigi.FloatParameter()
+
+    def get_task_name_str(self):
+        id_str = 'learn'
+        id_str += '_{}'.format(self.dataset)
```

Comment:
Umm, how about no? Create an array and join with `'_'.join(...)`

Reply:
This method has the virtue of being very flexible to change and update. Is that not reason enough to leave it like this?